### DISCUSSION TITLE
two different videos in a single project

### DISCUSSION THREAD

**Discussion by druoz** [talmolab/sleap#1311] [Help!]
Hello all, 
I have two questions, I would really appreciate if you reply.

1- When I work on two different videos in a single project, SLEAP gives the data as two different files. What are the advantages of combining two videos in one project instead of working on separate projects?

2- I removed the distortion in two videos. But as you can see in the picture I have two different viewports. I ran one of these in SLEAP. Can I combine these two videos and train them in SLEAP? Can I use the parameters in a single project and will the image difference be an issue? I want to measure distances between animals and between animals and frame corners. Is it possible to do this?

Thanks.

Ugur.

![screenshot undistort pic differences](https://github.com/talmolab/sleap/assets/127691619/e29cffee-7e31-4231-9491-35ea61ea2762)

**Comment by roomrys** [comment]
Hi @druoz,

Projects per Video
---
While SLEAP can resume the training of a model, it only uses labeled frames in the current project as example for further training. The advantage of **combining the videos into a single project** would be that you can use examples from both videos to train a model (**generalizing the model** to your videos). Alternatively, if you had separate projects, you would be training multiple "final" models (one for each project) where the "final" models will be tuned to the video in that specific project.

SLEAP works best when the videos being used are similar. Ideally, if the videos are similar enough, you can just train a model on one of your videos and have it output good predictions on the rest of your (unlabeled) videos. If there are some differences in the videos (as seen above), then you should include labels from both videos in one SLEAP project to create a generalized model that works on both.

My recommendation would be to add the new video to the existing SLEAP project (saved as a new version!) and add labels to the new video. It would be ideal if you had a single model, but if the model is having trouble generalizing, you can separate the projects by video via the [API](https://sleap.ai/api/sleap.io.dataset.html#sleap.io.dataset.Labels.remove_video):
```python
import sleap

ds = "path/to/slp"
labels = sleap.load_file(ds)

# Remove the first video (and all it's labels) from the labels project
video_to_remove = labels.videos[0]
labels.remove_video(video_to_remove)

new_ds = "new/path/to/slp"  # Use a new path/filename!
labels.save(new_ds)
```
No time would be wasted labeling as you would still be able to use the split labels to train a video-specific model if the generalized one does not pan-out.

Measuring Distances
---
>I am assuming the "frame" is not just the corner of the image (which we can measure the distance to the animal via the x,y-coordinates of our node locations), but either the cage corners or the black-to-image corners.

As for the goal of measuring distances between animals/corner-frames, you have two options:
1. Create a monster skeleton that encompasses both the animal and the frame
2. Create two separate projects/models (one containing a skeleton for the animals and another that has a skeleton for the frame)
I would recommend option 2 as the frame-project will require much fewer labels since it is a stationary object and should in general be easier. 
All the distance measurements would be done via the [analysis.h5](https://sleap.ai/notebooks/Analysis_examples.html) files (which are output per video).

Thanks,
Liezl
